### ISSUE TITLE
Error in Intra edge filter strength selection process

### ISSUE THREAD

**Issue by felixf3** [AOMediaCodec/av1-spec#270]
In Section 7.11.2.9, it is impossible to get “strength=0” because the spec is missing some terms that are in the reference code.

AV1 Spec: 
strength = 0
if ( filterType == 0 ) {
  if ( blkWh <= 8 ) {
    if ( d >= 56 ) strength = 1
  } else if ( blkWh <= 12 ) {
    if ( d >= 40 ) strength = 1
  } else if ( blkWh <= 16 ) {
    if ( d >= 40 ) strength = 1
  } else if ( blkWh <= 24 ) {
    if ( d >= 8 ) strength = 1
    if ( d >= 16 ) strength = 2
    if ( d >= 32 ) strength = 3
  } else if ( blkWh <= 32 ) {
    strength = 1
    if ( d >= 4 ) strength = 2
    if ( d >= 32 ) strength = 3
  } else {
    strength = 3
  }
} else {
  if ( blkWh <= 8 ) {
    if ( d >= 40 ) strength = 1
    if ( d >= 64 ) strength = 2
  } else if ( blkWh <= 16 ) {
    if ( d >= 20 ) strength = 1
    if ( d >= 48 ) strength = 2
  } else if ( blkWh <= 24 ) {
    if ( d >= 4 ) strength = 3
  } else {
    strength = 3
  }
}

Reference code:
  int strength = 0;

  const int blk_wh = bs0 + bs1;
  if (type == 0) {
    if (blk_wh <= 8) {
      if (d >= 56) strength = 1;
    } else if (blk_wh <= 12) {
      if (d >= 40) strength = 1;
    } else if (blk_wh <= 16) {
      if (d >= 40) strength = 1;
    } else if (blk_wh <= 24) {
      if (d >= 8) strength = 1;
      if (d >= 16) strength = 2;
      if (d >= 32) strength = 3;
    } else if (blk_wh <= 32) {
      if (d >= 1) strength = 1;
      if (d >= 4) strength = 2;
      if (d >= 32) strength = 3;
    } else {
      if (d >= 1) strength = 3;
    }
  } else {
    if (blk_wh <= 8) {
      if (d >= 40) strength = 1;
      if (d >= 64) strength = 2;
    } else if (blk_wh <= 16) {
      if (d >= 20) strength = 1;
      if (d >= 48) strength = 2;
    } else if (blk_wh <= 24) {
      if (d >= 4) strength = 3;
    } else {
      if (d >= 1) strength = 3;
    }
  }
  return strength;

**Comment by peterderivaz** [comment]
I assume you are referring to the case when d is equal to 0?

In this case, the specification does not invoke the intra filter strength selection process (see section 7.11.2.4 "If pAngle is not equal to 90 and pAngle is not equal to 180, the following applies").

So instead of filtering with a strength of zero, instead the specification doesn't apply the filter at all.  The idea is to make it clearer that no filtering is applied in the 90 and 180 degree cases.

**Comment by felixf3** [comment]
Yes, I’m referring to the case where d is 0 (pAngle is 90 or 180). If 7.11.2.9 is never invoked in that case, it leaves 7.11.2.9 somewhat confusing. It says "The output is an intra edge filter strength from 0 to 3 inclusive”, but zero cannot be its output. "strength=0” is set with the top line, but the if/else statement tree below it is a full case statement, and none of the possible outcomes are zero.

If the goal is to make is clear that 7.11.2.9 is only entered "If pAngle is not equal to 90 and pAngle is not equal to 180”, I would recommend the following changes to 7.11.2.9 for clarity:

	1. Change the wording to "The output is an intra edge filter strength from 1 to 3 inclusive.”
	2.  Remove the first line of code — “strength = 0”, since it serves no purpose

**Comment by peterderivaz** [comment]
Yes, I agree that would make sense as an editorial improvement.
